### PR TITLE
[Mobile Payments] Specific alert provider for built-in reader connections

### PR DESCRIPTION
**Pull request by joshheald**
Part of: #8082 

## Description

In sTAP Away, we're adding support for Apple's built in card readers to take in-person payments. We continue to use Stripe's SDK to handle payments.

Stripe's connection flow is technically very similar to the bluetooth reader flow, but conceptually, there are differences and it's important that we make it clear to users that they are using the built in reader, and don't give the wrong impression that they are connecting to a bluetooth reader, to avoid confusion.

This PR adds an alert provider specialised for connection to a built-in reader, along with new alert view models and temporary text and images for the scanning and connecting stages of the connecting flow.

It does not handle error states, or "software updates" – these will be done in a future PR.

There are also new protocols, splitting up the alerts so that when we refactor the two connection controllers to share duplicated code, they can work with the alert providers as opaque types.

## Testing instructions

Using an iPhone (XS or above on iOS 15.4 or newer, with a passcode set and signed in to iCloud)

1. Navigate to `Menu > Settings > Experimental features`
2. Turn on `Tap to Pay on iPhone`
3. Navigate to `Menu > Payments > Collect payment`
4. Go through the payment flow, and select `Card` on the payment method screen
5. When asked for a reader type, tap `Tap to Pay on iPhone` and go through the Terms of Service Apple ID linking (if you've not done so before)
6. Observe that specific messages for the built in reader are shown during the connecting process (note that images and text are all placeholders!)

## Screenshots

https://user-images.githubusercontent.com/2472348/205371624-1190c6fe-13f6-47c1-b020-62915ab2e9ad.mp4

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalBuiltInConnectingToReader.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import UIKit
+
+/// Modal presented when we are connecting to a reader
+///
+final class CardPresentModalBuiltInConnectingToReader: CardPresentPaymentsModalViewModel {
```
Comment (toupper) [inline WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalBuiltInConnectingToReader.swift:5]: A non-blocker suggestion here, I would make these namings easier to understand, perhaps `CardPresentBuiltInConnectingToReaderModalViewModel` and `CardPresentBuiltInReaderCheckingDeviceSupportModalViewModel`? I see they are long, so I leave it to your discernment.

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderConnectionAlertsProviding.swift [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+protocol MultipleCardReaderConnectionAlertsProviding {
+    /// Defines an interactive alert indicating a reader has been found. The user must
+    /// choose to connect to that reader or continue searching
+    ///
+    func foundReader(name: String,
+                     connect: @escaping () -> Void,
+                     continueSearch: @escaping () -> Void,
+                     cancelSearch: @escaping () -> Void) -> CardPresentPaymentsModalViewModel
+
+    // TODO: implement this approach, allowing us to remove the several readers logic from CardPresentPaymentAlertsPresenter
```
Comment (toupper) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderConnectionAlertsProviding.swift:64]: Commented code here?

**Comment by toupper** [comment]
The code looks good and it tests well, thanks for the PR @joshheald!

I observed one behavior when tapping to pay with my iPhone, it is not strictly related to this PR but It's worth mentioning. After paying with the card successfully and the native iOS modal is dismissed, we still show briefly the previous message that advises the user to present the card or retry, which is confusing to the user.
For instance in this case I had to retry the card tap in the native iOS modal. After it was successful with a second try and the modal was dismissed, we still showed "Retry Card" for a while:



If we pay successfully after one try, we show briefly the previous message:

**Review by toupper — APPROVED** [review]

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderConnectionAlertsProviding.swift (hunk 2 above)
Comment (joshheald) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderConnectionAlertsProviding.swift:64]: Yeah, it's deliberate, I'll be bringing it back in soon.

File: WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalBuiltInConnectingToReader.swift (hunk 1 above)
Comment (joshheald) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalBuiltInConnectingToReader.swift:5]: Makes sense. I'll put a note on the issue to update these when we have finalised strings and images for the screens: there's not really a lot of clarity over what the SDK is actually doing at each point, so naming them based on the message is probably the clearest approach.

**Comment by joshheald** [comment]
Thanks for the review @toupper 

> I observed one behavior when tapping to pay with my iPhone, it is not strictly related to this PR but It's worth mentioning. After paying with the card successfully and the native iOS modal is dismissed, we still show briefly the previous message that advises the user to present the card or retry, which is confusing to the user. For instance in this case I had to retry the card tap in the native iOS modal. After it was successful with a second try and the modal was dismissed, we still showed "Retry Card" for a while:
> 
> If we pay successfully after one try, we show briefly the previous message:

Yes, this is known... it's covered by #8289: I've not got to the payment flows yet, and this is an existing issue that is a lot more obvious when the phone is the reader 😊

**Pull request merged**